### ISSUE TITLE
Rule Change: add an option to ignore nested function definition in max-lines-per-function

### ISSUE THREAD

**Issue by yf-yang** [eslint/eslint#17741]
### What rule do you want to change?

max-lines-per-function

### What change to do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new option

### Example code

```js
function foo() {
  bar();
  bar();
  return;

  function bar() {}
}
```

### What does the rule currently do for this code?

Count the total lines of foo, including definition of bar

### What will the rule do after it's changed?

Add an additional option, so that function bar's definition does not count.

### Participation

- [ ] I am willing to submit a pull request to implement this change.

### Additional comments

This technique is quite common to create a scope to share parameters of the outer function to its nested functions without redeclare parameters again and again. Since the definition of the main function is not so long, then its nested function's lines won't influence its readability.

I am not so sure because you can also move the definition bar before the return statement, which do influence foo's readability. So instead of "ignore nested function definition", maybe "unreachable code in foo's body does not count for foo" is a better strategy.

**Comment by nzakas** [comment]
Thanks for the suggestion. I don't think adding an option makes sense, as this rule is already behaving the way it was intended. You can always create a custom version of the rule that does what you're looking for.
